### PR TITLE
Exception handler

### PR DESCRIPTION
**Pull request by KseniyaOsad**

### CODE REVIEW

File: OnlineLibrary.API/Controllers/BookController.cs [hunk 1]
```diff
@@ -30,102 +32,60 @@ public BookController(IBookService iBook, IAuthorService iAuthor, IMapper mapper
...
-            {
-                return NotFound(e.Message);
-            }
+            Book book = _mapper.Map<CreateBook, Book>(cBook);
+            book.Authors = authors;
+            return Ok(_bookService.CreateBook(book));
         }
 
         // PUT:  api/Book/UpdateReservation/[id]
         [HttpPut("{id}")]
```
Comment (an-lev) [inline OnlineLibrary.API/Controllers/BookController.cs:68]: https://www.geeksforgeeks.org/difference-between-put-and-patch-request/

Compare PUT and PATCH and decide what to use is better

File: OnlineLibrary.API/Controllers/BookController.cs [hunk 2]
```diff
@@ -30,102 +32,60 @@ public BookController(IBookService iBook, IAuthorService iAuthor, IMapper mapper
...
-                return NotFound(e.Message);
-            }
+            ExceptionHelper.Check<OLException>(Id != book.Id, "Id не совпадают", ExceptionType.BadRequest);
+            _bookService.ChangeBookReservation(Id, book.Reserve);
+            Book b = _bookService.GetBookById(Id);
+            return Ok(b);
         }
 
         // PUT:  api/Book/UpdateArchievation/[id]
         [HttpPut("{id}")]
```
Comment (an-lev) [inline OnlineLibrary.API/Controllers/BookController.cs:78]: Same as above

File: OnlineLibrary.Common/ErrorMiddleware/ErrorHandlerMiddleware.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
+﻿using Microsoft.AspNetCore.Http;
+using OnlineLibrary.Common.Exceptions;
```
Comment (an-lev) [inline OnlineLibrary.Common/ErrorMiddleware/ErrorHandlerMiddleware.cs:2]: think about what to use when middleware or filters 
In that case, it's also possible to use a global exception filter

File: OnlineLibrary.Common/Exceptions/OLException.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿using OnlineLibrary.Common.Exceptions.Enum;
+using System;
+
+namespace OnlineLibrary.Common.Exceptions
+{
+    public class OLException : Exception
```
Comment (an-lev) [inline OnlineLibrary.Common/Exceptions/OLException.cs:6]: Why OLException ?
Is it abbreviature ?

Comment (KseniyaOsad) [reply OnlineLibrary.Common/Exceptions/OLException.cs:6]: OLException == OnlineLibraryException

File: OnlineLibrary.Common/Filters/OLExceptionFilter.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+using System.Text;
+
+namespace OnlineLibrary.Common.Filters
+{
+    public class OLExceptionFilter : IExceptionFilter
+    {
+        public void OnException(ExceptionContext context)
+        {
+            Exception exc = context.Exception;
+            var olException = exc as OLException;
```
Comment (an-lev) [inline OnlineLibrary.Common/Filters/OLExceptionFilter.cs]: Let's rename to GenericExceptionFilter and leave OLException handling as is but add also default exception handling

File: OnlineLibrary.Common/Validators/BookValidator.cs [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+{
+    public class BookValidator : AbstractValidator<Book>
+    {
+        public BookValidator()
+        {
+            RuleFor(x => x.Name).NotEmpty();
+            RuleFor(x => x.Description).NotEmpty();
+            RuleFor(x => x.Authors).NotEmpty();
+            RuleFor(x => x.Genre).NotEmpty().IsInEnum();
+            //RuleFor(x => x.Tags).NotEmpty();
```
Comment (an-lev) [inline OnlineLibrary.Common/Validators/BookValidator.cs]: remove if it's useless

**Pull request merged**